docs/source/1.0/spec/core/idl.rst

```diff
@@ -0,0 +1,1886 @@
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        apply MyString @documentation("This is my string!")
+        apply MyString @length(min: 1, max: 10)
+
+    .. code-tab:: json
+
+        {
+            "smithy": "1.0",
+            "shapes": {
+                "smithy.example#MyString": {
+                    "type": "apply",
+                    "traits": {
+                        "smithy.api#documentation": "This is my string!",
+                        "smithy.api#length": {
+                            "min": 1,
+                            "max": 10
+                        }
+                    }
+                }
+            }
+        }
+
+Traits can be applied to members too:
+
+.. code-block:: smithy
+
+    namespace smithy.example
+
+    apply MyStructure$foo @documentation("Structure member documentation")
+    apply MyUnion$foo @documentation("Union member documentation")
+    apply MyList$member @documentation("List member documentation")
+    apply MySet$member @documentation("Set member documentation")
+    apply MyMap$key @documentation("Map key documentation")
+    apply MyMap$value @documentation("Map key documentation")
+
+.. seealso::
+
+    Refer to :ref:`trait conflict resolution <trait-conflict-resolution>`
+    for information on how trait conflicts are resolved.
+
+.. note::
+
+    In the semantic model, applying traits outside of a shape definition is
+    treated exactly the same as applying the trait inside of a shape
+    definition.
+
+
+.. _node-values:
+
+-----------
+Node values
+-----------
+
+*Node values* are analogous to JSON values. Node values are used to define
+:ref:`metadata <semantic-model>` and :ref:`trait values <traits>`.
+Smithy's node values have many advantages over JSON: comments,
+unquoted keys, unquoted strings, text blocks, and trailing commas.
+
+The following example defines a complex object metadata entry using a
+node value:
+
+.. code-block:: smithy
+
+    metadata foo = {
+        hello: 123,
+        "foo": "456",
+        testing: """
+            Hello!
+            """,
+        an_array: [10.5],
+        nested-object: {
+            hello-there$: true
+        }, // <-- Trailing comma
+    }
+
+.. rubric:: Array node
+
+A :token:`node_array` contains zero or more heterogeneous
+:token:`node_value`\s. An array node is defined like a JSON array: it is
+opened with a bracket (``[``), followed by zero or more ``node_value``\s,
+each value is separate by a comma (``,``), and closed by a bracket (``]``).
+A trailing comma is allowed in a node array.
+
+The following examples define arrays with zero, one, and two values:
+
+* ``[]``
+* ``[true]``
+* ``[1, "hello",]``
+
+.. rubric:: Object node
+
+A :token:`node_object` contains zero or more key value pairs of strings
+(a :token:`node_object_key`) that map to heterogeneous :token:`node_value`\s.
+An object node is defined like a JSON object: it is opened with a brace
```

Comment:
This written definition doesn't include a ":" for separating keys from their values.